edge/pkg/edged/edged.go

```diff
@@ -977,6 +992,93 @@ func (e *edged) syncPod() {
 	}
 }
 
+func (e *edged) handleVolume(op string, content []byte) (interface{}, error) {
+	switch op {
+	case constants.CSIOperationTypeCreateVolume:
+		return e.createVolume(content)
+	case constants.CSIOperationTypeDeleteVolume:
+		return e.deleteVolume(content)
+	case constants.CSIOperationTypeControllerPublishVolume:
+		return e.controllerPublishVolume(content)
+	case constants.CSIOperationTypeControllerUnpublishVolume:
+		return e.controllerUnpublishVolume(content)
+	}
+	return nil, nil
+}
+
+func (e *edged) createVolume(content []byte) (interface{}, error) {
+	req := &csi.CreateVolumeRequest{}
+	err := jsonpb.Unmarshal(bytes.NewReader(content), req)
+	if err != nil {
+		klog.Errorf("unmarshal create volume req error: %v", err)
+		return nil, err
+	}
+
+	klog.Infof("start create volume: %s", req.Name)
+	ctl := csiplugin.NewController()
+	res, err := ctl.CreateVolume(req)
+	if err != nil {
+		klog.Errorf("create volume error: %v", err)
+		return nil, err
+	}
+	klog.Infof("end create volume: %s result: %v", req.Name, res)
```

Comment:
do we need logs for start / end operations ? they don't convey any key information. if required move to V3 or V4 else debug logs

Reply:
Thanks. will use V4

Reply:
Done. Thank you